aws_adfs/html_roles_fetcher.py

```diff
@@ -35,6 +31,13 @@ def fetch_html_encoded_roles(
         password=None,
         sspi=True
 ):
+
```

Comment:
From the comments it appears this header should be conditional, only applied when SSPI is enabled.  In my environment (no SSPI) adfs returns a 401 when this header is sent.  Moving it here fixes that.